packages/dependency-extraction-webpack-plugin/lib/util.js

```diff
@@ -71,13 +73,19 @@ function defaultRequestToExternal( request ) {
  */
 function defaultRequestToExternalModule( request ) {
 	if ( request === '@wordpress/interactivity' ) {
-		// This is a special case. Interactivity does not support dynamic imports at this
-		// time. We add the external "module" type to indicate that webpack should
-		// externalize this as a module (instead of our default `import()` external type)
-		// which forces @wordpress/interactivity imports to be hoisted to static imports.
+		// This is a special case. Interactivity does not support dynamic imports at
+		// this time. We add the external "module" type to indicate that webpack
+		// should externalize this as a module (instead of our default `import()`
+		// external type) which forces @wordpress/interactivity imports to be
+		// hoisted to static imports.
 		return `module ${ request }`;
 	}
 
+	if ( request === '@wordpress/interactivity/router' ) {
+		// Assumes this is usually giong to be used as a dynamic import.
```

Comment:
```suggestion
		// Assumes this is usually going to be used as a dynamic import.
```

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/57924/commits/c15686cd6e956e556b20c861db1f1a810eba11bd.